### ISSUE TITLE
Allow routing through unpaved roads even with unpaved roads disabled when starting position is on an unpaved road

### ISSUE THREAD

**Issue by AlertSubject** [organicmaps/organicmaps#3285]
_This issue was created based on the post [here](https://talk.ahoxus.org/notice/AMnjpmiRSMRFlGDubo) as the user in question doesn't have a Github account_

Quote: 

"go to an unpaved route. check "avoid unpaved route". try routing anywhere. it doesn't work because every route will go through unpaved!
now go to the middle of a highway with tolls. check "avoid tolls". try routing. it will give you the shortest way to get out of it.
even if you don't check "avoid tolls", routing will show where they stand on the route.
unpaved roads should work the same as tolls. avoid means to take the alternative whenever possible, but i still want to see the options."

**Comment by biodranik** [comment]
Right, avoid should not mean "fail with the error if only that type of road is available".
